imports.go

```diff
@@ -58,19 +59,33 @@ type Importer interface {
 }
 
 // Contents is a representation of imported data. It is a simple
-// string wrapper, which makes it easier to enforce the caching policy.
+// byte wrapper, which makes it easier to enforce the caching policy.
 type Contents struct {
-	data *string
+	data *[]byte
 }
 
 func (c Contents) String() string {
+	// Construct string without copying underlying bytes.
+	// NB: This only works because c.data is not modified.
+	return *(*string)(unsafe.Pointer(c.data))
+}
+
+func (c Contents) Data() []byte {
 	return *c.data
 }
 
 // MakeContents creates Contents from a string.
 func MakeContents(s string) Contents {
+	data := []byte(s)
```

Comment:
It suddenly occurred to me that this probably breaks the 'repeating MakeContents(foo) produces the exact same Contents' rule.  Might need to change the Contents representation to store both a []byte pointer and a string pointer, or improve the 'is the same' check.

Reply:
False alarm?  I think there's no problem here, and we're good to proceed.

---

More info: I think we get a different `Contents` even when called with the same string, ie:
```golang
foo := "foo"
a, b := MakeContents(foo), MakeContents(foo)
a != b
```
... because of the above line of code ... but afaics this is ok.

Afaics the intended semantics ([enforced](https://github.com/google/go-jsonnet/blob/fb6c700f29d1bcc6b0be96f042670094c69a7001/imports.go#L111-L113) by `importCache.importData()`) are that `Importer.Import` returns a copy of the _same_ `Contents` object for the same `foundAt` value -- not a new `Contents` object made from the same string (subtle difference).

ie: Importer.Import() is meant to do this:
```golang
// Good example
func (i *myimporter) ImportData(from, path string) (contents Contents, foundAt string, err error) {
        foundAt = i.resolvePath(from, path)
        if contents = i.cache[foundAt]; contents == nil {
                contents = MakeContents(i.readContents(foundAt))
                // Important! Return (a copy of) the same `Contents` for a given `foundAt`
                i.cache[foundAt] = contents
        }
        return
}
```

Not this:

```golang
// Wrong/bad example
func (i *myimporter) ImportData(from, path string) (contents Contents, foundAt string, err error) {
        foundAt = i.resolvePath(from, path)
        var stringData string
        if stringData, ok = i.dataCache[foundAt]; !ok {
                stringData = i.readContents(foundAt)
                i.dataCache[foundAt] = stringData
        }
        // Wrong! Now we have a 'new' Contents object for the same `foundAt`
        contents = MakeContents(stringData)
        return
}

Reply:
Yes, exactly. We enforce that it's the same Contents object. 

(It encourages efficient caching by avoiding unnecessary copies and we can quickly compare pointers for equality instead of going over the whole string).